integration/samples/typings/specs/typings.ts

```diff
@@ -0,0 +1,33 @@
+import { expect } from 'chai';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+describe(`sample-typings`, () => {
+  describe(`validator.d.ts`, () => {
+    let VALIDATOR_DTS;
+    before(() => {
+      VALIDATOR_DTS = fs.readFileSync(
+        path.resolve(__dirname, '..', 'dist', 'validator.d.ts'), 'utf-8');
+    });
+
+    it(`should triple-slash reference to 'src/chalk.d.ts'`, () => {
+      /// <reference path="src/chalk.d.ts" />
+      expect(VALIDATOR_DTS).to.contain('src/chalk.d.ts');
+      expect(VALIDATOR_DTS).to.satisfy(
+        (value: string) => value.startsWith(`/// <reference path="src/chalk.d.ts" />`),
```

Comment:
Use Case: include additional typings in compilation, write such typings to distributable bundle

Underlying social conflict: Customer lock-in to third-party on version x. Vendor lock-in who won't ship types prior to version x+(something). If possible, one could ship type definitions on an internal npm registry (why are we not doing it?!?)

Proposed solution: setting `compilerOptions.outDir` equal to `compilerOptions.basePath` will output `*.{js,js.map,d.ts}` in the same folder as input `*.ts` files. In a subsequent step, copy `<basePath>/**/*.d.ts` to the distributable bundle - et voila, triple-slash referenced typings are included. In v1, was implemented by copying files on disk (to a temporary working directory).